internal/rpc/user_devices.go

```diff
@@ -639,24 +641,119 @@ func (s *userDeviceRPCServer) UpdateUserDeviceMetadata(ctx context.Context, req
 }
 
 func (s *userDeviceRPCServer) ClearMetaTransactionRequests(ctx context.Context, _ *emptypb.Empty) (*pb.ClearMetaTransactionRequestsResponse, error) {
-	currTime := time.Now()
-	fifteenminsAgo := currTime.Add(-time.Minute * 15)
 
-	m, err := models.MetaTransactionRequests(
-		models.MetaTransactionRequestWhere.CreatedAt.LT(fifteenminsAgo),
-		qm.OrderBy(models.MetaTransactionRequestColumns.CreatedAt+" ASC"),
+	conn, err := grpc.Dial(s.settings.MetaTransactionProcessorGRPCAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to meta transaction processor: %w", err)
+	}
+
+	client := mtpgrpc.NewMetaTransactionServiceClient(conn)
+	// call to meta transaction connector to clear the transaction and get id
+
+	response, err := client.CleanStuckMetaTransactions(ctx, &emptypb.Empty{})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to clear meta transaction: %w", err)
+	}
+
+	metaTransaction, err := models.MetaTransactionRequests(
+		models.MetaTransactionRequestWhere.ID.EQ(response.Id),
+		models.MetaTransactionRequestWhere.Status.NEQ("Confirmed"),
+		models.MetaTransactionRequestWhere.Hash.IsNull(),
+		qm.Load(models.MetaTransactionRequestRels.MintRequestVehicleNFT),
+		qm.Load(models.MetaTransactionRequestRels.BurnRequestVehicleNFT),
+		qm.Load(models.MetaTransactionRequestRels.MintRequestSyntheticDevice),
+		qm.Load(models.MetaTransactionRequestRels.BurnRequestSyntheticDevice),
+		qm.Load(models.MetaTransactionRequestRels.ClaimMetaTransactionRequestAftermarketDevice),
+		qm.Load(models.MetaTransactionRequestRels.PairRequestAftermarketDevice),
+		qm.Load(models.MetaTransactionRequestRels.UnpairRequestAftermarketDevice),
 	).One(ctx, s.dbs().Reader)
+
 	if err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
 			return nil, fmt.Errorf("no overdue meta transaction")
 		}
 		return nil, fmt.Errorf("failed to select transaction to clear: %w", err)
 	}
 
-	_, err = m.Delete(ctx, s.dbs().Writer)
+	if metaTransaction.R != nil {
+
+		if metaTransaction.R.MintRequestVehicleNFT != nil {
+			vehicleNft := metaTransaction.R.MintRequestVehicleNFT
+
+			vehicleNft.MintRequestID = ""
+			_, err = vehicleNft.Update(ctx, s.dbs().Writer, boil.Infer())
+			if err != nil {
+				return nil, fmt.Errorf("failed to update vehicleNft %s: %w", vehicleNft.TokenID, err)
+			}
+		}
+
+		if metaTransaction.R.BurnRequestVehicleNFT != nil {
+			vehicleNft := metaTransaction.R.BurnRequestVehicleNFT
+
+			vehicleNft.BurnRequestID = null.StringFrom("")
```

Comment:
is there an option for null string, or null.New, maybe even null.StringFromPtr(nil)